CometServer/Modules/10-25/EngineeringModelApi.cs

```diff
@@ -212,6 +212,185 @@ public override void AddRoutes(IEndpointRouteBuilder app)
+        /// <param name="permissionInstanceFilterService">The service for filtering permission instances.</param>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        private async Task GetAll(HttpRequest request, HttpResponse response, string ids, IRequestUtils requestUtils, ICdp4TransactionManager transactionManager, 
+            ICredentialsService credentialsService, IHeaderInfoProvider headerInfoProvider, Services.IServiceProvider serviceProvider, 
+            IMetaInfoProvider metaInfoProvider, ICdp4JsonSerializer jsonSerializer, 
+            IMessagePackSerializer messagePackSerializer, IPermissionInstanceFilterService permissionInstanceFilterService)
+        {
+            if (!(await this.IsServerReady(response)))
+            {
+                return;
+            }
+
+            if (!(request.HttpContext.User.Identity?.IsAuthenticated ?? false))
+            {
+                response.UpdateWithNotAuthenticatedSettings();
+                await response.AsJson("not authenticated");
+            }
+            else
+            {
+                try
+                {
+                    await this.Authorize(this.AppConfigService, credentialsService, request.HttpContext.User.Identity.Name);
+                }
+                catch (AuthorizationException)
+                {
+                    this.logger.LogWarning("The GET REQUEST was not authorized for {identity}", request.HttpContext.User.Identity.Name);
+                    response.UpdateWithNotAutherizedSettings();
+                    await response.AsJson("not authorized");
+                    return;
+                }
+                
+                NpgsqlConnection connection = null;
+                var credentials = credentialsService.Credentials;
+                var transaction = transactionManager.SetupTransaction(ref connection, credentials);
+
+                transactionManager.SetCachedDtoReadEnabled(true);
+                transactionManager.SetFullAccessState(true);
+
+                var stopwatch = Stopwatch.StartNew();
+                var requestToken = this.TokenGeneratorService.GenerateRandomToken();
+
+                this.logger.LogInformation("{request}:{requestToken} - START HTTP REQUEST PROCESSING", request.QueryNameMethodPath(), requestToken);
+                
+                try
+                {
+                    var engineeringModels = new List<Thing>();
+
+                    var processor = new ResourceProcessor(transaction, serviceProvider, requestUtils, metaInfoProvider);
+
+                    var allEngineeringModelIds = new List<Guid>();
+
+                    if (ids is null)
+                    {
+                        requestUtils.OverrideQueryParameters = new QueryParameters() { ExtentDeep = true };
+
+                        allEngineeringModelIds.AddRange(this.ProcessRequestPath(requestUtils, transactionManager, processor, nameof(SiteDirectory), nameof(SiteDirectory),
```

Comment:
please have a look at the `DetermineEngineeringModelSetup`, there we first get the SiteDirectory, then the contained EngineeringModelSetups. could we not reuse the same design pattern. Because with the proposed method here we get the full siteDirectory which is too much data and too much processing